File: content-solutions/dfs/club-count.java
import java.util.*;

class Solution {
    private List<List<Integer>> graph;
    private boolean[] visited;

    void dfs(int v) {
        visited[v] = true;
        for (int w : graph.get(v)) {
            if (!visited[w]) dfs(w);
        }
    }

    public int solution(int n, int[][] pairs) {
        graph = new ArrayList<>();
        for (int i = 0; i < n; i++) graph.add(new ArrayList<>());
        for (int[] e : pairs) {
            graph.get(e[0]).add(e[1]);
            graph.get(e[1]).add(e[0]);
        }
        visited = new boolean[n];
        int count = 0;
        for (int v = 0; v < n; v++) {
            if (!visited[v]) {
                count++;
                dfs(v);
            }
        }
        return count;
    }
}

File: content-solutions/dfs/club-count.js
function solution(n, pairs) {
  const graph = Array.from({ length: n }, () => []);
  for (const [a, b] of pairs) {
    graph[a].push(b);
    graph[b].push(a);
  }
  const visited = new Array(n).fill(false);
  function dfs(v) {
    visited[v] = true;
    for (const w of graph[v]) {
      if (!visited[w]) dfs(w);
    }
  }
  let count = 0;
  for (let v = 0; v < n; v += 1) {
    if (!visited[v]) {
      count += 1;
      dfs(v);
    }
  }
  return count;
}
